File: ccif/src/java/other/org/himalay/ccif/ServiceIPUnicast.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:55 EST 2016")

public  class ServiceIPUnicast extends BinStruct implements PublicBinMsg{




    // members variables
    // macAddress
    public MacAddress macAddress ;
    // optionaData
    public ByteArray optionaData ;

        public ServiceIPUnicast () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize macAddress
        macAddress = new MacAddress();
        // Initialize optionaData
        optionaData= new ByteArray();optionaData.setSizeType("FIRST_UI8");
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read macAddress
        retVal += macAddress.read(istream);
        // read optionaData
        {retVal+=optionaData.read(istream); }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        { /** fix dependent sizes for macAddress **/  }
        { /** fix dependent sizes for optionaData**/  }
    
                // write macAddress
        if (macAddress!=null)retVal +=macAddress.write(ostream);
        // write optionaData
        {retVal += optionaData.write(ostream);}
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("ServiceIPUnicast\n");
    dc.increaseIndent();
        int retVal= 0;
        // write macAddress
        if ( macAddress != null ) {dc.indent();dc.getPs().println("macAddress") ;retVal +=macAddress.dump(dc);}
        // write optionaData
        dc.indent();dc.getPs().print("optionaData: "+optionaData.getSize()+"(0x"+Integer.toHexString(optionaData.getSize())+")\n");this.optionaData.dump(dc);
dc.decreaseIndent();
        return retVal;
    }


        // Getter for macAddress
    //public MacAddress getMacAddress()
    //{
    //    return macAddress ;
    //}

    
    // Setter for macAddress
    //public void setMacAddress(MacAddress val)
    //{
    //    this.macAddress= val;
    //}
    // Getter for optionaData
    //public ByteArray getOptionaData()
    //{
    //    return optionaData ;
    //}

    
    // Setter for optionaData
    //public void setOptionaData(ByteArray val)
    //{
    //    this.optionaData= val;
    //}


    public void setOptionaData(byte[] val)
    {
        this.optionaData.setData(val);
    }
    
    
    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/ServiceIPUnicast.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:12 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "ServiceIPUnicast.h"

#ifdef org_himalay_ccif_ServiceIPUnicast__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
ServiceIPUnicast::ServiceIPUnicast() {
    references= 0;


    // macAddress
    org_himalay_ccif_MacAddress__NEW( macAddress);
    // optionaData
    org_himalay_msgs_runtime_ByteArray_NEW(optionaData);
    optionaData->setSizeType("FIRST_UI8");
}
// Destructor
ServiceIPUnicast::~ServiceIPUnicast() {
    // macAddress
    org_himalay_ccif_MacAddress__DELETE( macAddress);
    // optionaData
    org_himalay_msgs_runtime_ByteArray_DELETE( optionaData);
}

int ServiceIPUnicast::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // macAddress
    retVal += macAddress->read(istream);
    // optionaData
    {
        retVal+=optionaData->read(istream);
    }
    return retVal;
}

int ServiceIPUnicast::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read macAddress
    retVal += macAddress->read(istream);
    // read optionaData
    {
        retVal+=optionaData->read(istream);
    }

    return retVal;
}

int ServiceIPUnicast::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for macAddress **/
    }
    {   /** fix dependent sizes for optionaData**/
    }

    int retVal= 0;
    // write macAddress
    if ( macAddress != NULL ) {
        retVal +=macAddress->write(ostream);
    }
    // write optionaData
    {
        retVal += optionaData->write(ostream);
    }

    return retVal;
}

int ServiceIPUnicast::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("ServiceIPUnicast\n");
    dc.increaseIndent();
    int retVal= 0;
    // write macAddress
    if ( macAddress != NULL ) {
        dc.indent();
        dc.getPs().println("macAddress") ;
        retVal +=macAddress->dump(dc);
    }
    // write optionaData
    dc.indent();
    dc.getPs().print("optionaData");
    optionaData->dump(dc);
    dc.decreaseIndent();
    return retVal;
}


// Getter for macAddress
//org_himalay_ccif_MacAddress__PTR_TYPE ServiceIPUnicast::getMacAddress()
//{
//return this->macAddress;
//}

// Setter for macAddress
//void ServiceIPUnicast::setMacAddress(org_himalay_ccif_MacAddress__PTR_TYPE val)
//{
//this->macAddress= val;
//}

// Getter for optionaData
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE ServiceIPUnicast::getOptionaData()
//{
//return this->optionaData;
//}

// Setter for optionaData
//void ServiceIPUnicast::setOptionaData(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->optionaData= val;
//}

int ServiceIPUnicast::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_ServiceIPUnicast__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::ServiceIPUnicast* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::ServiceIPUnicast* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code